sdk/src/wallet/account/operations/syncing/mod.rs

```diff
@@ -162,66 +162,36 @@ where
         // the new output ids
         let mut new_alias_and_nft_addresses = HashMap::new();
-        let (mut spent_or_not_synced_output_ids, mut addresses_with_unspent_outputs, mut outputs_data) =
-            (Vec::new(), Vec::new(), Vec::new());
 
-        loop {
-            let new_outputs_data = if new_alias_and_nft_addresses.is_empty() {
-                // Get outputs for addresses and add them also the the addresses_with_unspent_outputs
-                let (addresses_with_output_ids, spent_or_not_synced_output_ids_inner) = self
-                    .get_output_ids_for_addresses(options, addresses_to_sync.clone())
-                    .await?;
-                spent_or_not_synced_output_ids = spent_or_not_synced_output_ids_inner;
-                // Get outputs for addresses and add them also the the addresses_with_unspent_outputs
-                let (addresses_with_unspent_outputs_inner, outputs_data_inner) = self
-                    .get_outputs_from_address_output_ids(addresses_with_output_ids)
-                    .await?;
-                addresses_with_unspent_outputs = addresses_with_unspent_outputs_inner;
-                outputs_data.extend(outputs_data_inner.clone());
-                outputs_data_inner
-            } else {
-                let bech32_hrp = self.client().get_bech32_hrp().await?;
-                let mut new_outputs_data = Vec::new();
-                for (alias_or_nft_address, ed25519_address) in new_alias_and_nft_addresses {
-                    let output_ids = self.get_output_ids_for_address(alias_or_nft_address, options).await?;
-
-                    // Update address with unspent outputs
-                    let address_with_unspent_outputs = addresses_with_unspent_outputs
-                        .iter_mut()
-                        .find(|a| a.address.inner == ed25519_address)
-                        .ok_or_else(|| {
-                            crate::wallet::Error::AddressNotFoundInAccount(ed25519_address.to_bech32(bech32_hrp))
-                        })?;
-                    address_with_unspent_outputs.output_ids.extend(output_ids.clone());
-
-                    let new_outputs_data_inner = self.get_outputs(output_ids).await?;
-
-                    let outputs_data_inner = self
-                        .output_response_to_output_data(new_outputs_data_inner, address_with_unspent_outputs)
-                        .await?;
-
-                    outputs_data.extend(outputs_data_inner.clone());
-                    new_outputs_data.extend(outputs_data_inner);
-                }
-                new_outputs_data
-            };
+        let bech32_hrp = self.client().get_bech32_hrp().await?;
+
+        let mut new_outputs_data = outputs_data.clone();
 
+        loop {
             // Clear, so we only get new addresses
-            new_alias_and_nft_addresses = HashMap::new();
+            new_alias_and_nft_addresses.clear();
```

Comment:
We can just move the creation here `let mut new_alias_and_nft_addresses = HashMap::new();` and then don't need the clear anymore?

Reply:
no, because then we would have to re-allocate in each iteration. This way ... we only allocate once.

Reply:
oh, we can just drain this as well

Reply:
Done